Sources/AWSSDKSwift/Extensions/DynamoDB/DynamoDB+Codable.swift

```diff
@@ -0,0 +1,273 @@
+                expressionAttributeValues: input.expressionAttributeValues,
+                item: item,
+                returnConsumedCapacity: input.returnConsumedCapacity,
+                returnItemCollectionMetrics: input.returnItemCollectionMetrics,
+                returnValues: input.returnValues,
+                tableName: input.tableName
+            )
+            return putItem(request, on: eventLoop)
+        } catch {
+            let eventLoop = eventLoop ?? client.eventLoopGroup.next()
+            return eventLoop.makeFailedFuture(error)
+        }
+    }
+
+    /// The GetItem operation returns a set of attributes for the item with the given primary key. If there is no matching item, GetItem does not return any data and there will be no Item element in the response.  GetItem provides an eventually consistent read by default. If your application requires a strongly consistent read, set ConsistentRead to true. Although a strongly consistent read might take more time than an eventually consistent read, it always returns the last updated value.
+    public func getItemCodable<T: Decodable>(_ input: GetItemInput, type: T.Type, on eventLoop: EventLoop? = nil) -> EventLoopFuture<GetItemCodableOutput<T>> {
+        return getItem(input, on: eventLoop)
+            .flatMapThrowing { response -> GetItemCodableOutput<T> in
+                let item = try response.item.map { try DynamoDBDecoder().decode(T.self, from: $0) }
+                return GetItemCodableOutput(
+                    consumedCapacity: response.consumedCapacity,
+                    item: item
+                )
+        }
+    }
+    
+    /// Edits an existing item's attributes, or adds a new item to the table if it does not already exist. You can put, delete, or add attribute values. You can also perform a conditional update on an existing item (insert a new attribute name-value pair if it doesn't exist, or replace an existing name-value pair if it has certain expected attribute values). You can also return the item's attribute values in the same UpdateItem operation using the ReturnValues parameter.
+    public func updateItemCodable<T: Encodable>(_ input: UpdateItemCodableInput<T>, on eventLoop: EventLoop? = nil) -> EventLoopFuture<UpdateItemOutput> {
+        do {
+            var item = try DynamoDBEncoder().encode(input.updateItem)
+            // extract key from input object
+            var key: [String: AttributeValue] = [:]
+            input.key.forEach {
+                key[$0] = item[$0]!
+                item[$0] = nil
+            }
+            // construct expression attribute name and value arrays from name attribute value map
+            let expressionAttributeNames: [String: String] = .init( item.keys.map { ("#\($0)", $0) }) { first,_ in return first }
+            let expressionAttributeValues: [String: AttributeValue] = .init( item.map { (":\($0.key)", $0.value) }) { first,_ in return first }
```

Comment:
`expressionAttributeValues` could be the default property of UpdateItemCodableInput

Reply:
see above

Reply:
Not sure why you would want this, kinda removes the requirement for the supplied Codable `updateItem`